### PR TITLE
vulkan test coverage for 1d, 3d, 3dlist images

### PR DESCRIPTION
**Pull request by pkaur19**

### CODE REVIEW

**Comment by shajder** [comment]
@pkaur19 thanks for this contribution. I am not talkin in behalf of `OpenCL` work group but as someone who recently went across missing coverage of `vulkan` interoperability. As result I've created [this](https://github.com/KhronosGroup/OpenCL-CTS/issues/1834) issue. That being said, I am not sure if `OpenCL-CTS` needs to test list of 3D images (`VulkanImage3DList` class), it sounds to me a bit like time consuming operation meanwhile many vendors already complain about CTS execution time. Would that make any sense to replace list of `image3d_t` test with `image1d_array_t` / `image_2d_array_t` tests ? 

BTW. I went briefly through your code and I can't spot any location of construction of `VulkanImage3DList` object, am I missing something? Is this work in progress?

Please sign CLA license to move forward with your PR, thanks.

File: test_conformance/common/vulkan_wrapper/vulkan_list_map.cpp [hunk 1]
```diff
@@ -319,6 +315,57 @@ VulkanImage2DList::~VulkanImage2DList()
...
+VulkanImage3DList::VulkanImage3DList(
+    size_t numImages, std::vector<VulkanDeviceMemory *> &deviceMemory,
+    uint64_t baseOffset, uint64_t interImageOffset, const VulkanDevice &device,
+    VulkanFormat format, uint32_t width, uint32_t height, uint32_t depth, uint32_t mipLevels,
+    VulkanImageTiling vulkanImageTiling,
+    VulkanExternalMemoryHandleType externalMemoryHandleType,
+    VulkanImageCreateFlag imageCreateFlag, VulkanImageUsage imageUsage,
+    VulkanSharingMode sharingMode)
+{
+    for (size_t i2DIdx = 0; i2DIdx < numImages; i2DIdx++)
```
Comment (nikhiljnv) [inline test_conformance/common/vulkan_wrapper/vulkan_list_map.cpp:333]: @shajder Were you looking for this?

**Comment by nikhiljnv** [comment]
@shajder 
The purpose of the test is to add the test coverage you requested as part of #1834, not for addition to generic image test coverage. This is the work-in-progress, but I have left a comment to point you to part you are referring to. Also, I would prefer to get some basic functional changes in and follow up with another PR as we keep adding more coverage.

**Comment by pkaur19** [comment]
> @pkaur19 thanks for this contribution. I am not talkin in behalf of `OpenCL` work group but as someone who recently went across missing coverage of `vulkan` interoperability. As result I've created [this](https://github.com/KhronosGroup/OpenCL-CTS/issues/1834) issue. That being said, I am not sure if `OpenCL-CTS` needs to test list of 3D images (`VulkanImage3DList` class), it sounds to me a bit like time consuming operation meanwhile many vendors already complain about CTS execution time. Would that make any sense to replace list of `image3d_t` test with `image1d_array_t` / `image_2d_array_t` tests ?
> 
> BTW. I went briefly through your code and I can't spot any location of construction of `VulkanImage3DList` object, am I missing something? Is this work in progress?
> 
> Please sign CLA license to move forward with your PR, thanks.

Sure if 3DList is not needed I will replace 3DList with 1DList. AlsoI see `VulkanImage2DList` is already present we don't need to implement that.

VulkanImage3DList is created [here](https://github.com/KhronosGroup/OpenCL-CTS/pull/1883/commits/3a532ef4983837211ba252a08abfca9f6e5c2efe#diff-fe4c0fb9e87097b3eaed07e7ed7e71b13c19ba358c44a4c424e9e9f5e33e6a11R235).

File: test_conformance/common/vulkan_wrapper/vulkan_wrapper.cpp [hunk 2]
```diff
@@ -2278,8 +2325,6 @@ VulkanSemaphore::VulkanSemaphore(
     vkCreateSemaphore(m_device, &vkSemaphoreCreateInfo, NULL, &m_vkSemaphore);
 }
 
-const VulkanDevice &VulkanSemaphore::getDevice() const { return m_device; }
```
Comment (joshqti) [inline test_conformance/common/vulkan_wrapper/vulkan_wrapper.cpp:2281]: Why delete this?

File: test_conformance/common/vulkan_wrapper/vulkan_wrapper.cpp [hunk 3]
```diff
@@ -2325,23 +2370,6 @@ int VulkanSemaphore::getHandle(
 
         return fd;
     }
-    else if (externalSemaphoreHandleType
-             == VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_SYNC_FD)
-    {
```
Comment (joshqti) [inline test_conformance/common/vulkan_wrapper/vulkan_wrapper.cpp:2330]: Why delete this code

**Review by joshqti — CHANGES_REQUESTED** [review]
I think there may be some rebase stuff that snuck in that needs fixing.  I would like to test this, on our platform before it merges.

**Comment by joshqti** [comment]
I cannot build this change.  Can you please ensure is builds with CI?

**Comment by nikhiljnv** [comment]
Closing this in favor of https://github.com/KhronosGroup/OpenCL-CTS/pull/1887. Please, review https://github.com/KhronosGroup/OpenCL-CTS/pull/1887 instead.

**Pull request closed without merging**